File: 1-two-sum/two-sum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val a = mutableMapOf<Int,Int>()
        for(i in 0 until nums.size){
            if(a[target-nums[i]] == null) a[nums[i]] = i
            else return intArrayOf(a.getOrDefault(target-nums[i],0),i)
        }
        return intArrayOf(-1,-1)
    }
}

File: 1-two-sum/two-sum.dart
class Solution {
  List<int> twoSum(List<int> nums, int target) {
    var map = Map<int,int>();
    int i = 0;
    while(i<nums.length){
        if(map[target-nums[i]] != null) return [map[target-nums[i]]!,i];
        else map[nums[i]] = i;
        i++;
    }
    return [-1,-1];
  }
}